pkg/codegen/go/gen_program_expressions.go

```diff
@@ -986,9 +986,7 @@ func (g *generator) literalKey(x model.Expression) (string, bool) {
 				break
 			}
 		}
-		var buf bytes.Buffer
-		g.GenTemplateExpression(&buf, x)
-		return buf.String(), true
+		return "", false
```

Comment:
The result of GenTemplateExpression will never be a literal; it only will be a literal in the case already handled on L984.